src/renderer/components/PerCurrencySelectAccount/index.js

```diff
@@ -0,0 +1,214 @@
+
+const mapStateToProps = createStructuredSelector({
+  accounts: shallowAccountsSelector,
+});
+
+type Option = {
+  matched: "boolean",
+  account: Account,
+  subAccount: SubAccount,
+};
+const getOptionValue = option => {
+  return option.account ? option.account.id : null;
+};
+
+const defaultFilter = createFilter({
+  stringify: ({ data: account }) => {
+    const currency = getAccountCurrency(account);
+    const name = getAccountName(account);
+    return `${currency.ticker}|${currency.name}|${name}`;
+  },
+});
+const filterOption = o => (candidate, input) => {
+  const selfMatches = defaultFilter(candidate, input);
+  if (selfMatches) return [selfMatches, true];
+  return [false, false];
+};
+
+const AccountOption = React.memo(function AccountOption({
+  account,
+  subAccount,
+  currency,
+  isValue,
+}: {
+  account: Account,
+  subAccount: SubAccount,
+  currency: CryptoCurrency | TokenCurrency,
+  isValue?: boolean,
+}) {
+  return <MenuOption isValue={isValue} account={account} subAccount={subAccount} />;
+});
+
+type OwnProps = {
+  currency: TokenCurrency | CryptoCurrency,
+  mandatoryTokens?: boolean,
+  value: ?AccountLike,
+  onChange: (account: ?AccountLike, tokenAccount: ?Account) => void,
+};
+
+type Props = OwnProps & {
+  accounts: Account[],
+  t: TFunction,
+  onChange: (account: ?AccountLike, tokenAccount: ?Account) => void, // I have to add it twice else flow is unhappy, don't ask why
+};
+
+const getAccountsByCurrency = (accounts: Account[], currency: TokenCurrency | CryptoCurrency) =>
+  accounts.filter((acc: Account) => acc.currency.id === currency.id);
+
+// TODO: token account generation logic in live-common
+const generateTokenAccount = (account: Account, currency: TokenCurrency) => ({
+  type: "TokenAccount",
+  id: account.id + "+" + currency.contractAddress,
+  parentId: account.id,
+  token: currency,
+  balance: BigNumber(0),
+  operationsCount: 0,
+  creationDate: new Date(),
+  operations: [],
+  pendingOperations: [],
+  starred: false,
+});
+
+const RawSelectAccount = ({
+  accounts,
+  onChange,
+  t,
+  currency,
+  mandatoryTokens,
+  ...props
+}: Props) => {
+  const availableAccounts = useMemo(() => {
+    if (currency.type === "TokenCurrency") {
+      const mainAccounts = getAccountsByCurrency(accounts, currency.parentCurrency);
+      return mainAccounts.map((acc: Account) => {
+        return {
+          account: acc,
+          subAccount:
+            (acc.subAccounts &&
+              acc.subAccounts.find(
+                (subAcc: SubAccount) =>
+                  subAcc.type === "TokenAccount" && subAcc.token.id === currency.id,
+              )) ||
+            generateTokenAccount(acc, currency),
+        };
+      });
+    }
+    return getAccountsByCurrency(accounts, currency).map((acc: Account) => ({
+      account: acc,
+      subAccount: null,
+    }));
+  }, [accounts, currency]);
+
+  const [selectedValue, setSelectedValue] = useState();
+
+  useEffect(() => {
```

Comment:
That effect is dangerous because it assumes the accounts prop never update.

This needs to use a ref to statically store the initial value because I think you want it to be just at init?

```
const initialAcc = useRef(availableAccounts)
```